diff --git a/src/ft2font_wrapper.cpp b/src/ft2font_wrapper.cpp
index 98e9ff74f..102cfbda3 100644
--- a/src/ft2font_wrapper.cpp
+++ b/src/ft2font_wrapper.cpp
@@ -637,8 +637,9 @@ static PyObject *PyFT2Font_set_text(PyFT2Font *self, PyObject *args, PyObject *k
     if (PyUnicode_Check(textobj)) {
         size = PyUnicode_GET_LENGTH(textobj);
         codepoints.resize(size);
+        Py_UNICODE *unistr = PyUnicode_AsUnicode(textobj);
         for (size_t i = 0; i < size; ++i) {
-            codepoints[i] = PyUnicode_ReadChar(textobj, i);
+            codepoints[i] = unistr[i];
         }
     } else if (PyBytes_Check(textobj)) {
         size = PyBytes_Size(textobj);
