diff --git a/setupext.py b/setupext.py
index 2b1fd9349c..4fe30ec55e 100644
--- a/setupext.py
+++ b/setupext.py
@@ -633,7 +633,7 @@ class FreeType(SetupPackage):
                 "--with-png=no", "--with-harfbuzz=no", "--enable-static",
                 "--disable-shared"
             ]
-            host = sysconfig.get_config_var('BUILD_GNU_TYPE')
+            host = sysconfig.get_config_var('HOST_GNU_TYPE')
             if host is not None:  # May be unset on PyPy.
                 configure.append(f"--host={host}")
             subprocess.check_call(configure, env=env, cwd=src_path)
