diff --git a/src/_c_internal_utils.c b/src/_c_internal_utils.c
index 4a61fe5b6e..c1949811d4 100644
--- a/src/_c_internal_utils.c
+++ b/src/_c_internal_utils.c
@@ -68,7 +68,8 @@ mpl_GetCurrentProcessExplicitAppUserModelID(PyObject* module)
     wchar_t* appid = NULL;
     HRESULT hr = GetCurrentProcessExplicitAppUserModelID(&appid);
     if (FAILED(hr)) {
-        return PyErr_SetFromWindowsErr(hr);
+        PyErr_SetFromWindowsErr(hr);
+        return NULL;
     }
     PyObject* py_appid = PyUnicode_FromWideChar(appid, -1);
     CoTaskMemFree(appid);
@@ -89,7 +90,8 @@ mpl_SetCurrentProcessExplicitAppUserModelID(PyObject* module, PyObject* arg)
     HRESULT hr = SetCurrentProcessExplicitAppUserModelID(appid);
     PyMem_Free(appid);
     if (FAILED(hr)) {
-        return PyErr_SetFromWindowsErr(hr);
+        PyErr_SetFromWindowsErr(hr);
+        return NULL;
     }
     Py_RETURN_NONE;
 #else

