--- matplotlib-3.1.0.orig/setupext.py	2019-05-17 22:55:28.000000000 -0300
+++ matplotlib-3.1.0/setupext.py	2019-05-18 08:49:36.376442717 -0300
@@ -830,7 +830,7 @@
             ext, 'libpng',
             atleast_version='1.2',
             alt_exec=['libpng-config', '--ldflags'],
-            default_libraries=['png', 'z'])
+            default_libraries=['libpng', 'z'])
         Numpy().add_flags(ext)
         return ext
 
